### PR TITLE
[Instrumentation.ElasticsearchClient] Fix sensitive data in url.full (former db.url) tag

### LINKED ISSUE #1194: Elasticsearch db.statement and db.url tags contain authentication details.
# Issue with OpenTelemetry.Instrumentation.ElasticsearchClient

`db.statement` and `db.url` contains an http link that is authenticated `https://user:pass@host/....`

List of [all OpenTelemetry NuGet
packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are
using (e.g. `OpenTelemetry 1.3.2`):

* 1.4.0 RTM all other packages.
* OpenTelemetry.Instrumentation.ElasticsearchClient 1.0.0-beta4

Runtime version (e.g. `net462`, `net48`, `net6.0`, `net7.0` etc. You can
find this information from the `*.csproj` file):

* net6.0

**Is this a feature request or a bug?**

* [ ] Feature Request
* [x] Bug

**What is the expected behavior?**

I'd expect user and passwords to be redacted.

**What is the actual behavior?**

I see the http url contain both user name and password.

### PR DESCRIPTION
**Pull request by soalexmn**
Fixes #1194

## Changes

This PR should partially close https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1194

Remove sensitive information (username and password) in `url.full` ( former `db.url`) as described in https://opentelemetry.io/docs/specs/semconv/database/elasticsearch/

Plus, adjust semantic conventions, use url.full instead of db.url

I'm not sure about `db.statement` tag as soon as library redact password itself (checked with my traces) and cleaning such data in a large query may affect the query data itself in case of false positives.

### CODE REVIEW

**Comment by soalexmn** [comment]
> @soalexmn, 2 important thigs
> 
>     1. Please sign CLA. Without this we cannot proceed with PR.
> 
>     2. `db.url` seems to be kind of custom attribute. Could you please adjust semantic convention first, then make appropriate sanitisation?

Thanks, I've corrected all the points

**Review by ejsmith — APPROVED** [review]

File: src/Shared/SemanticConventions.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ internal static class SemanticConventions
     public const string AttributeDbStatement = "db.statement";
     public const string AttributeDbOperation = "db.operation";
     public const string AttributeDbInstance = "db.instance";
-    public const string AttributeDbUrl = "db.url";
+    public const string AttributeDbUrl = "url.full";
```
Comment (Kielek) [inline src/Shared/SemanticConventions.cs:56]: This attribute is used only by Elasticsearch package. It should be fine to rename it, but please replace also the C# const name. Without this, you are breaking convention used in the whole file.

```suggestion
    public const string AttributeUrlFull = "url.full";
```

**Review by Kielek — COMMENTED** [review]
Please add also CHANGELOG.md entry. Inform about 2 things
1. Replacing old attribute by the new one (include names)
2. Redacting values.

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@
   ([#1624](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1624))
 * Update OpenTelemetry SDK version to `1.8.1`.
   ([#1668](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1668))
+* Fix sensitive data in `url.full` (former `db.url`) tag
+  ([#1684](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1684))
+  * Replace `db.url` attribute with `url.full` to comply with conventions.
+  * Fix instrumentation to replace username and password in `url.full` with `REDACTED:REDACTED`.
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.ElasticsearchClient/CHANGELOG.md:14]:
```suggestion
* Replace `db.url` attribute with `url.full` to comply with [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/v1.25.0/docs/database/elasticsearch.md#attributes).
  Redact `username` and `password` part of the `url.full`.
  ([#1684](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1684))
```

**Comment by soalexmn** [comment]
I apologize for the slight mix up, the build should be going well now. Should I squash the commits?

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
> @Kielek I apologize for the slight mix up, the build should be going well now. Should I squash the commits?

There is no need. We are squashing while merging to the main.

**Pull request merged**